src/DetailsView/components/issue-filing-dialog.tsx

```diff
@@ -1,77 +1,138 @@
-                    primaryButtonHref={href}
-                    primaryButtonText={'File issue'}
+    public render(): JSX.Element {
+        const { selectedBugData, onClose, isOpen, deps } = this.props;
+        const { selectedBugFilingService } = this.state;
+        const selectedBugFilingServiceData = this.state.selectedBugFilingService.getSettingsFromStoreData(
+            this.state.bugServicePropertiesMap,
+        );
+        const environmentInfo = deps.environmentInfoProvider.getEnvironmentInfo();
+        const isSettingsValid = selectedBugFilingService.isSettingsValid(selectedBugFilingServiceData);
+        const href = isSettingsValid
+            ? selectedBugFilingService.issueFilingUrlProvider(selectedBugFilingServiceData, selectedBugData, environmentInfo)
+            : null;
+
+        return (
+            <Dialog
+                className={'issue-filing-dialog'}
+                hidden={!isOpen}
+                dialogContentProps={{
+                    type: DialogType.normal,
+                    title: titleLabel,
+                    titleId: 'issue-filing-dialog-title',
+                    subText: 'This configuration can be changed again in settings.',
+                    subTextId: 'issue-filing-dialog-subtext',
+                    showCloseButton: false,
+                }}
+                modalProps={{
+                    isBlocking: false,
+                    containerClassName: 'insights-dialog-main-override',
+                }}
+                onDismiss={onClose}
+            >
+                <BugFilingSettingsContainer
+                    deps={deps}
+                    selectedBugFilingService={selectedBugFilingService}
+                    selectedBugFilingServiceData={selectedBugFilingServiceData}
+                    onPropertyUpdateCallback={this.onPropertyUpdateCallback}
+                    onSelectedServiceChange={this.onSelectedServiceChange}
                 />
-            </DialogFooter>
-        </Dialog>
-    );
-});
+                <DialogFooter>
+                    <ActionAndCancelButtonsComponent
+                        isHidden={false}
+                        primaryButtonDisabled={isSettingsValid === false}
+                        primaryButtonOnClick={this.onPrimaryButtonClick}
+                        cancelButtonOnClick={onClose}
+                        primaryButtonHref={href}
+                        primaryButtonText={'File issue'}
+                    />
+                </DialogFooter>
+            </Dialog>
+        );
+    }
+
+    private onPrimaryButtonClick = (ev: React.SyntheticEvent<Element, Event>) => {
+        const newData = this.state.selectedBugFilingService.getSettingsFromStoreData(this.state.bugServicePropertiesMap);
+        const service = this.state.selectedBugFilingService.key;
+        this.props.deps.userConfigMessageCreator.saveIssueFilingSettings(service, newData);
+        this.props.bugFileTelemetryCallback(ev);
```

Comment:
would it make sense to move the telemetry part to be the responsibility of `saveIssueFilingSettings` or what ever code handle that message on the background?

Reply:
Since save does not necessarily mean file, I would like not couple for the moment.